core/trino-main/src/main/java/io/trino/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -268,7 +270,14 @@ private TableHandle findTableScanHandle(PlanNode node)
                 return findTableScanHandle(((SemiJoinNode) node).getSource());
             }
             if (node instanceof JoinNode) {
-                return findTableScanHandle(((JoinNode) node).getLeft());
+                JoinNode joinNode = (JoinNode) node;
+                return findTableScanHandle(joinNode.getLeft());
```

Comment:
redundant change in "Support uncorrelated subquery in UPDATE assignments (SET)".
 inline `joinNode` back.